packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -308,49 +308,50 @@ export class GraphQLTransform {
   }
 
   public applyOverride = (stackManager: StackManager): AmplifyApiGraphQlResourceStackTemplate => {
-    const stacks: string[] = [];
-    const amplifyApiObj: any = {};
-    stackManager.rootStack.node.findAll().forEach(node => {
-      const resource = node as CfnResource;
-      if (resource.cfnResourceType === 'AWS::CloudFormation::Stack') {
-        stacks.push(node.node.id.split('.')[0]);
-      }
-    });
+    const overrideFilePath = path.join(this.overrideConfig!.overrideDir, 'build', 'override.js');
+    if (!_.isEmpty(this.overrideConfig) && this.overrideConfig!.overrideFlag && fs.existsSync(overrideFilePath)) {
```

Comment:
The fix is in this file. I.e. there's added check if override file exists and condition is pulled to the top of method to spare some useless computing.